src/practice4.tsx

```diff
@@ -0,0 +1,105 @@
+import React, { useEffect } from 'react';
+import { SubmitHandler, useForm } from 'react-hook-form';
+import axios from 'axios';
+import styles from './components/Practice4.module.scss';
+
+type FormData = {
+  zipcode: string;
+  city: string;
+  street: string;
+};
+
+const Practice4 = () => {
+  const {
+    register,
+    watch,
+    setValue,
+    setError,
+    clearErrors,
+    resetField,
+    handleSubmit,
+    formState: { errors },
+  } = useForm<FormData>();
+  const onSubmit: SubmitHandler<FormData> = (data) => console.log(data);
+
+  useEffect(() => {
+    const zipcodeValue = watch('zipcode');
```

Comment:
watchよりuseWatchの方がいいと思います。

> Behaves similarly to the watch API, however, this will isolate re-rendering at the custom hook level and potentially result in better performance for your application.

https://react-hook-form.com/docs/usewatch